src/analysis/special_functions/gamma.lean

```diff
@@ -1214,3 +1218,259 @@ end
+    apply this,
+    rw neg_lt,
+    rcases lt_or_le 0 (re s) with hs | hs,
+    { exact (neg_neg_of_pos hs).trans_le (nat.cast_nonneg _), },
+    { refine (nat.lt_floor_add_one _).trans_le _,
+      rw [sub_eq_neg_add, nat.floor_add_one (neg_nonneg.mpr hs), nat.cast_add_one] } },
+  intro m,
+  induction m with m IH generalizing s,
+  { -- Base case: `0 < re s`, so Gamma is given by the integral formula
+    intro hs,
+    rw [nat.cast_zero, neg_zero] at hs,
+    rw [←Gamma_eq_Gamma_aux],
+    { refine tendsto.congr' _ (approx_Gamma_integral_tendsto_Gamma_integral hs),
+      refine (eventually_ne_at_top 0).mp (eventually_of_forall (λ n hn, _)),
+      exact (Gamma_seq_eq_approx_Gamma_integral hs hn).symm },
+    { rwa [nat.cast_zero, neg_lt_zero] } },
+  { -- Induction step: use recurrence formulae in `s` for Gamma and Gamma_seq
+    intro hs,
+    rw [nat.cast_succ, neg_add, ←sub_eq_add_neg, sub_lt_iff_lt_add, ←one_re, ←add_re] at hs,
+    rw Gamma_aux,
+    have := tendsto.congr' ((eventually_ne_at_top 0).mp (eventually_of_forall (λ n hn, _)))
+      ((IH _ hs).div_const s),
+    swap 3, { exact Gamma_seq_add_one_left s hn }, -- doesn't work if inlined?
+    conv at this in (_ / _ * _) { rw mul_comm },
+    rwa [←mul_one (Gamma_aux m (s + 1) / s), tendsto_mul_iff_of_ne_zero _ (one_ne_zero' ℂ)] at this,
+    simp_rw add_assoc,
+    exact tendsto_coe_nat_div_add_at_top (1 + s) }
+end
+
+end complex
+
+end limit_formula
+
+section gamma_reflection
+/-! ## The reflection formula -/
+
+open_locale real
+namespace complex
+
+lemma Gamma_seq_mul (z : ℂ) {n : ℕ} (hn : n ≠ 0) :
+  Gamma_seq z n * Gamma_seq (1 - z) n =
+  n / (n + 1 - z) * (1 / (z * ∏ j in finset.range n, (1 - z ^ 2 / (j + 1) ^ 2))) :=
+begin
+  -- also true for n = 0 but we don't need it
+  have aux : ∀ (a b c d : ℂ), a * b * (c * d) = a * c * (b * d), by { intros, ring },
+  rw [Gamma_seq, Gamma_seq, div_mul_div_comm, aux, ←pow_two],
+  have : (n : ℂ) ^ z * n ^ (1 - z) = n,
+  { rw [←cpow_add _ _ (nat.cast_ne_zero.mpr hn), add_sub_cancel'_right, cpow_one] },
+  rw [this, finset.prod_range_succ', finset.prod_range_succ, aux, ←finset.prod_mul_distrib,
+    nat.cast_zero, add_zero, add_comm (1 - z) n, ←add_sub_assoc],
+  have : ∀ (j : ℕ), (z + ↑(j + 1)) * (1 - z + ↑j) = ↑((j + 1) ^ 2) * (1 - z ^ 2 / (↑j + 1) ^ 2),
+  { intro j,
+    push_cast,
+    have : (j:ℂ) + 1 ≠ 0, by { rw [←nat.cast_succ, nat.cast_ne_zero], exact nat.succ_ne_zero j },
+    field_simp, ring },
+  simp_rw this,
+  rw [finset.prod_mul_distrib, ←nat.cast_prod, finset.prod_pow,
+    finset.prod_range_add_one_eq_factorial, nat.cast_pow,
+    (by {intros, ring} : ∀ (a b c d : ℂ), a * b * (c * d) = a * (d * (b * c))),
+    ←div_div, mul_div_cancel, ←div_div, mul_comm z _, mul_one_div],
+  exact pow_ne_zero 2 (nat.cast_ne_zero.mpr $ nat.factorial_ne_zero n),
+end
+
+/-- At `-n` for `n ∈ ℕ`, the Gamma function is undefined; by convention we assign it the value 0. -/
+lemma Gamma_neg_nat_eq_zero (n : ℕ) : Gamma (-n) = 0 :=
```

Comment:
Can you move this one to another section where it would fit better? It's a general elementary fact on Gamma, not needing the reflection formula.